src/vs/platform/actions/common/actions.ts

```diff
@@ -477,7 +480,9 @@ export class MenuItemAction implements IAction {
 	run(...args: any[]): Promise<void> {
 		let runArgs: any[] = [];
 
-		if (this._options?.arg) {
+		if (this._options?.arg && Array.isArray(this._options.arg)) {
+			runArgs = [...runArgs, ...this._options.arg];
+		} else if (this._options?.arg) {
```

Comment:
I have unfortunately limited capacities for a deeper look but this seems breaking for those cases where use have been using an array as argument, right?